### ISSUE TITLE
Append channelToken automatically during local development

### ISSUE THREAD

**Issue by tcaruth** [oracle/content-management-sdk#8]
I'm using the content sdk through the OCM toolkit, and my graphql calls are failing because there isn't a channel token set as a query parameter. I'll submit a pull request with a possible solution, but feel free to let me know if there's a preferred way to do this.

**Comment by kmacdonaO** [comment]
@tcaruth (Sorry for not seeing this earlier)

When you say you're using the SDK through the OCM Toolkit, can you explain what you're calling on the toolkit, or are you directly calling the SDK that is bundled with the toolkit?

The SDK has a beforeSend function which is usually how I'd recommend customizing any requests you need to make but I'm not sure whether you have direct access to this as I'm not that familiar with the toolkit.

**Comment by tcaruth** [comment]
The OCM toolkit has the sdk bundled with it and uses that instead of pulling it from npm
https://github.com/oracle/content-and-experience-toolkit/blob/master/sites/test/server/npm/contentSDK.js

An OCM component gets passed the SitesSDK (and other things) into it's constructor that I can use to get access to a contentSDK that has the content delivery or content preview client already instantiated. This means I don't need to set a channel or context up for it, it's just available for use immediately.

@BrianJCheyne may be able to provide more insight, I'm only semi-familiar with how the toolkit forwards requests to OCM

**Comment by kmacdonaO** [comment]
Can you show your code that is invoking the method on the delivery client?

**Comment by kmacdonaO** [comment]
Is your problem that the contentClient does not have a channelToken, or what is it that is missing?  A channelToken can be given as you are in the graphQL query, or it can be added as anHTTP header (e.g. via a beforeSend added to the API call).

**Comment by tcaruth** [comment]
As I understand it, the toolkit acts as a passthrough, or proxy. It takes calls to OCM's content APIs on localhost and points them at the server specified in the command `cec develop -s targetServer`.

When I tried using the contentSDK.graphql locally I was recieving an error stating I the channel token wasn't defined. I tested and found that to get a response, I had to provide channelToken either as a header or query parameter to get a response through the toolkit proxy. I can get on a Zoom and demonstrate if you like

I could take the provided delivery/preview client passed into my component's constructor, read it's properties and call createDeliveryClient again with a beforeSend, but that seems convoluted to me.

I think this ticket is more about how OCM and the toolkit passes the contentSDK to our components. I could file a ticket over there if it seems more appropriate.
